src/tests/Common/CLRTest.NativeAot.targets

```diff
@@ -0,0 +1,191 @@
+fi
+        ]]>
+    </NativeAotCleanupBashScript>
+
+      <NativeAotBashScript Condition="'$(CLRTestKind)' == 'BuildAndRun' and '$(NativeAotTest)' != 'false'">
+        <![CDATA[
+if [ ! -z ${RunNativeAot+x} ]%3B then
+    __Command=$_DebuggerFullPath
+    # Tests run locally need __TestDotNetCmd (set by runtest.py) or a compatible 5.0 dotnet runtime in the path
+    if [ ! -z ${__TestDotNetCmd+x} ] %3B then
+        __Command+=" $__TestDotNetCmd"
+    else
+        __Command+=" dotnet"
+    fi
+
+    __Command+=" msbuild /p:IlcPath=$CORE_ROOT/nativeaot nativebuild.proj"
+    echo "Running Native AOT: $__Command"
+    $__Command
+    __naotExitCode=$?
+
+    if [ $__naotExitCode -ne 0 ]; then
+      echo Native AOT failed with exitcode: $__naotExitCode
+      exit 1
+    fi
+
+    # Copy the native component to the directory where the executable resides.
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        cp *.dylib  native/ 2>/dev/null
+    else
+        cp *.so  native/ 2>/dev/null
+    fi
+
+    ExePath=native/$(MSBuildProjectName)
+fi
+        ]]>
+      </NativeAotBashScript>
+
+      <BashCLRTestPreCommands>$(NativeAotCleanupBashScript);$(BashCLRTestPreCommands);$(NativeAotBashScript)</BashCLRTestPreCommands>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="GetNativeAotBatchScript">
+    <PropertyGroup>
+      <NativeAotMultimoduleIncompatibleScript Condition="'$(NativeAotMultimoduleIncompatible)' == 'true'">
+        <![CDATA[
+  ECHO SKIPPING EXECUTION BECAUSE TEST IS NOT COMPATIBLE WITH MULTIMODULE COMPILATION
+  popd
+  Exit /b 0
+        ]]>
+      </NativeAotMultimoduleIncompatibleScript>
+
+      <NativeAotCleanupBatchScript Condition="'$(CLRTestKind)' == 'BuildAndRun' and '$(NativeAotTest)' != 'false'">
+        <![CDATA[
+if defined RunNativeAot (
+    rem Delete any leftover native compilation bits
+    IF EXIST native rmdir /s /q native
+)
+        ]]>
+    </NativeAotCleanupBatchScript>
+
+    <NativeAotBatchScript Condition="'$(CLRTestKind)' == 'BuildAndRun' and '$(NativeAotTest)' != 'false'">
+        <![CDATA[
+if defined RunNativeAot (
+
+    set __Command=!_DebuggerFullPath!
+    REM Tests run locally need __TestDotNetCmd (set by runtest.py) or a compatible 5.0 dotnet runtime in the path
+    if defined __TestDotNetCmd (
+        set __Command=!__Command! "!__TestDotNetCmd!"
+    ) else (
+        set __Command=!__Command! "dotnet"
+    )
+
+    set __Command=!__Command! msbuild /p:IlcPath=!CORE_ROOT!\nativeaot nativebuild.proj
+
+    if defined NativeAotMultimodule (
+        set __Command=!__Command! /p:IlcMultiModule=true /p:DisableFrameworkLibGeneration=true /p:FrameworkLibPath=!CORE_ROOT!\nativeaot\sdk
+        $(NativeAotMultimoduleIncompatibleScript)
+    )
+
+    echo "!__Command!"
+    call !__Command!
+
+    if errorlevel 1 (
+        ECHO END COMPILATION - FAILED
+        ECHO FAILED
+        rem TODO: Why doesn't "exit /b" end with failure? Crossgen uses that above, but it results in success here!?
```

Comment:
If this comment is still valid, we should probably rename Crossgen to NativeAOT, otherwise it looks like an overlooked copy-paste.

Reply:
The comment refers to these lines in the crossgen2 targets file:

https://github.com/dotnet/runtime/blob/03194f790b7ce24c60e5727c5394b7df09f978e4/src/tests/Common/CLRTest.CrossGen.targets#L259-L262

When I had `exit /b` here, AOT compilation failure would result in a test success. It was very puzzling. Without the `/b`, AOT compilation failure properly fails the test, but also closes the command prompt if ran manually outside the xunit runner so this is rather annoying.

I'll retest if this still works that way. If so, I'll double check what it does in crossgen2's case since there's the `/b` that I saw as problematic.

Reply:
Lol, okay, so I spent the whole morning troubleshooting this.

If you have this batch file:

```batch
(
  exit /b 5
)
echo Hello
```

and run it with

```csharp
using System;
using System.Diagnostics;


Process process = new Process();
process.StartInfo.FileName = @"C:\git\runtime\artifacts\tests\coreclr\windows.x64.Debug\nativeaot\SmokeTests\BasicThreading\BasicThreading\aa.cmd";
process.StartInfo.UseShellExecute = false;

process.Start();
process.WaitForExit();

int exitCode = process.ExitCode;
Console.WriteLine(exitCode);
```

The `Console.WriteLine` will print exit code 5. So far so good. Makes sense. This is what I would expect.

Now move the `echo Hello` inside the block so that you have:

```batch
(
  exit /b 5
  echo Hello
)
```

Now it prints `0`. Funfunfunfunfun.